hipscat/partitioner.py

```diff
@@ -488,11 +370,25 @@ def write_structure_metadata(self):
 
         if self.debug:
             print(dumped_metadata)
-
-
-    def construct_hips_structure(self):
-        assert self.opix is not None, 'bad'
-        self.hips_structure = self.opix
+    
+
+    def write_parquet_metadata(self, hips_dir='catalog'):
+        print(f'Writing parquet metadata for hipscat {hips_dir}s')
+        root_dir = os.path.join(self.output_dir, hips_dir)
+        tab = ps.dataset(root_dir, partitioning='hive', format='parquet')
+        mc = []
+        
+        for f in tab.files:
+            tmd = pq.read_metadata(f)
+            tmd.set_file_path(f.split(f'{hips_dir}/')[1])
+            mc.append(tmd)
+
+        _meta_data_path = os.path.join(root_dir, '_metadata')
+        _common_metadata_path = os.path.join(root_dir, '_common_metadata')
+
+        pq.write_metadata(tab.schema, _meta_data_path, metadata_collector=mc)
```

Comment:
This is surprisingly straightforward. Was not expecting that.

Reply:
yeah, just wasn't documented on how to populate the metadata_collector this way.